api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -53,6 +53,30 @@ public interface TableScan extends Scan<TableScan, FileScanTask, CombinedScanTas
    */
   TableScan asOfTime(long timestampMillis);
 
+  /**
+   * Create a new {@link TableScan} from this scan's configuration that will use the tip snapshot of the given branch
+   *
+   * @param branch a snapshot ID
+   * @return a new scan based on this with the given snapshot ID
+   * @throws IllegalArgumentException if the branch cannot be found
+   */
+  default TableScan useBranch(String branch) {
+    throw new UnsupportedOperationException("Scanning from a branch is not supported");
+  }
+
+  /**
+   * Create a new {@link TableScan} from this scan's configuration that will use the most recent snapshot
+   * as of the given time in milliseconds.
+   *
+   * @param branch branch name
+   * @param timestampMillis  a timestamp in milliseconds.
+   * @return a new scan based on this with the given branch and timestamp
+   * @throws IllegalArgumentException if the branch or snapshot cannot be found
+   */
+  default TableScan useBranch(String branch, long timestampMillis) {
```

Comment:
what about `asOfTimeAtBranch`?

Reply:
How about useBranchAsOfTime?

Reply:
Updated to useBranchAsOfTime. I think it's good ot have the use just because we already have the useSnapshot method.

Reply:
+1 for useBranchAsOfTime

Reply:
+1